apps/api/src/app/events/usecases/send-message/test-send-email.usecase.ts

```diff
@@ -65,25 +70,40 @@ export class SendTestEmail {
       })
     );
 
-    const mailData: IEmailOptions = {
-      to: Array.isArray(email) ? email : [email],
-      subject,
-      html: html as string,
-      from: command.payload.$sender_email || integration?.credentials.from || 'no-reply@novu.co',
-    };
-
     if (email && integration) {
-      await this.sendMessage(integration, mailData, mailFactory);
+      const mailData: IEmailOptions = {
+        to: Array.isArray(email) ? email : [email],
+        subject,
+        html: html as string,
+        from: command.payload.$sender_email || integration?.credentials.from || 'no-reply@novu.co',
+      };
+
+      await this.sendMessage(integration, mailData, mailFactory, command);
 
       return;
     }
   }
 
-  private async sendMessage(integration: IntegrationEntity, mailData: IEmailOptions, mailFactory: MailFactory) {
-    const mailHandler = mailFactory.getHandler(integration, mailData.from);
+  private async sendMessage(
+    integration: IntegrationEntity,
+    mailData: IEmailOptions,
+    mailFactory: MailFactory,
+    command: TestSendMessageCommand
+  ) {
+    const { providerId } = integration;
+    const mailHandler = mailFactory.getHandler(
+      { ...integration, providerId: GetNovuIntegration.mapProviders(ChannelTypeEnum.EMAIL, providerId) },
+      mailData.from
```

Comment:
I would move all this bit of code inside the `try` so in case of misconfiguration for the MailHandler or parsing error in the providers mapping we would still get signalled that the error was `Unexpected provider error`. Easier to identify at debugging as downstream we have no error identification.